src/plugins/minimal_scene/EngineToQtInterface.cc

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "EngineToQtInterface.hh"
+
+#include <ignition/rendering/Camera.hh>
+#include <ignition/rendering/RenderEngine.hh>
+
+#include <QOpenGLExtraFunctions>
+
+#include <GL/glcorearb.h>
+
+// clang-format off
+namespace ignition
+{
+namespace gui
+{
+namespace plugins
+{
+  /// \brief Private data class for IgnRenderer
+  class EngineToQtInterfacePrivate
+  {
+    /// \brief FBO texture. Stores uploaded GPU -> CPU.
+    /// Used only during fallback.
+    public: GLuint fallbackTexture = 0;
+
+    /// \brief Stores downloaded GPU -> CPU. Used only during fallback
+    public: ignition::rendering::ImagePtr fallbackImage;
+
+    public: QOpenGLContext *glContext = nullptr;
+  };
+}
+}
+}
+// clang-format on
+
+using namespace ignition;
+using namespace gui;
+using namespace plugins;
+
+/////////////////////////////////////////////////
+EngineToQtInterface::EngineToQtInterface(QOpenGLContext *_glContext)
+{
+  this->dataPtr.reset(new EngineToQtInterfacePrivate);
```

Comment:
Use `std::make_unique` instead. For ex: https://github.com/ignitionrobotics/ign-gui/blob/3753ef2455370e3479eb5fccffbcb04caa59d97e/src/plugins/point_cloud/PointCloud.cc#L101